app/controllers/application_controller.rb

```diff
@@ -418,6 +418,20 @@ def maintenance_warning
     @minutes_to_maintenance = NextMaintenance.minutes_to(now)
   end
 
+  # Respond to a MissingTemplate error with 404.
+  # This prevents the server from responding with a 500
+  # in cases where a template for a requested format is
+  # not found.
+  rescue_from(ActionView::MissingTemplate) do |e|
+    logger.info(
+      "An ActionView::MissingTemplate error occurred, responding with 404",
+      :missing_template,
+      requested_url: request.original_url
+    )
+    render "/errors/status_404", status: 404, content_type: "text/html", locals: { status: 404 }
```

Comment:
@mporkola Yes, another option would be just to respond with an empty 404 and not touch the content type. Or even return the html representation with some other content type, as the application works in normal cases if for example `.json` is requested.

@rap1ds This is a clear downside. The logging is added to track these but of course it does not notify us about the occurrences.